src/liberty/devCommands.ts

```diff
@@ -7,528 +7,595 @@
-        } else {
-            vscode.window.showQuickPick(items).then(selection => {
-                if (!selection) {
-                    return;
-                }
-                const yes = localize("confirmation.button.label.yes");
-                const no = localize("confirmation.button.label.no");
-                vscode.window
-                .showInformationMessage(localize("remove.custom.project.confirmation", Path.dirname(selection.detail)), yes, no)
-                .then(answer => {
-                    if (answer === yes) {
-                    // delete and save
-                        projectProvider.removeInPersistedProjects(selection.detail);
-                        vscode.window
-                            .showInformationMessage(localize("remove.custom.project.successful"));
-                        projectProvider.fireChangeEvent();
-                    }
-                });
-            });
-        }   
-    }
-}
-
-export async function addProject(uri: vscode.Uri): Promise<void> {
-    const projectProvider: ProjectProvider = ProjectProvider.getInstance();
-    if (undefined !== uri && undefined !== uri.fsPath ) {
-        // Right mouse clicked on a root folder, or on empty space with only one folder in workspace.
-        // Add project if:
-        // 1. Not in liberty dashboard
-        // 2. Project has build files (pom.xml or build.gradle)
-        // 
-        // Once added, presist the data in workspace storage.
-        const result: number = await projectProvider.addUserSelectedPath(uri.fsPath, projectProvider.getProjects());
-        const message = localize(`add.project.manually.message.${result}`, uri.fsPath);
-        (result!==0)? console.error(message):console.info(message);projectProvider.fireChangeEvent();
-        vscode.window.showInformationMessage(message);
-        
-    } else {
-        // clicked on the empty space and workspace has more than one folders, or
-        // from command palette
-        // Display the list of workspace folders for user to select.
-        // The list should not contain any existing projects
-        const uris: string[] = [];
-        const wsFolders = vscode.workspace.workspaceFolders;
-		if ( wsFolders ) {
-			for ( const folder of wsFolders ) {
-				const path = folder.uri.fsPath;
-				if ( projectProvider.projectRootPathExists(path, projectProvider.getProjects().keys() ) === false ) {
-					uris.push(folder.uri.fsPath);
-				}
-			}
-		}
-        if ( uris.length === 0 ) {
-            // show error
-            const message = localize("add.project.manually.no.projects.available.to.add");
-            console.error(message);
-            vscode.window.showInformationMessage(message);
-        } else {
-            // present the list
-            vscode.window.showQuickPick(uris).then(async selection => {
-                if (!selection) {
-                    return;
+        if ( paths.length === 1) {
```

Comment:
With this implementation we are always reading the port from the server.env file correct? Even with the default `7777`?

Reply:
correct